src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2094,7 +2154,32 @@ public void HandleIDebugEntryPointEvent2(IDebugEngine2 pEngine, IDebugProcess2 p
 
         public void HandleIDebugBreakpointEvent2(IDebugEngine2 pEngine, IDebugProcess2 pProcess, IDebugProgram2 pProgram, IDebugThread2 pThread, IDebugEvent2 pEvent)
         {
-            FireStoppedEvent(pThread, StoppedEvent.ReasonValue.Breakpoint);
+            IList<Tracepoint> tracepoints = GetTracepoints(pEvent as IDebugBreakpointEvent2);
+            if (tracepoints.Any())
+            {
+                ThreadPool.QueueUserWorkItem((o) =>
+                {
+                    foreach (var tp in tracepoints)
+                    {
+                        string logMessage = tp.GetLogMessage(pThread, Constants.EvaluationRadix, m_processName);
+
+                        m_logger.WriteLine(LoggingCategory.DebuggerStatus, logMessage);
+                    }
+                });
+            }
+
+            if (!m_isStepping && tracepoints.Any())
```

Comment:
>m_isStepping [](start = 17, length = 12)

Your idea here is that if we are stepping, stopping at the tracepoint is going to mess up the step, so we should probably stop? Probably worth a comment at least.

Reply:
Yep. Ran into a bug where we would get a step event. Fire the step event in the debugger. Debugger sends a BreakpointEvent and instead of stoppping for the step, we would continue.

Reply:
Got it. GDB also has other issues if you mix tracepoints and stepping, so sounds reasonable.